src/test/webapp/js/unit-tests.js

```diff
@@ -39,104 +39,157 @@ var errorHandler = {
 
-                        assert.deepEqual(
-                                doc.body.contents[0].contents[0].styleAttrs["http://www.w3.org/ns/ttml#styling fontSize"],
-                                {"unit": "%", "value": 10.5}
-                        );
+                assert.deepEqual(
+                    doc.body.contents[0].contents[0].styleAttrs["http://www.w3.org/ns/ttml#styling fontSize"],
+                    {"unit": "%", "value": 10.5}
+                );
 
-                        assert.deepEqual(
-                                doc.body.contents[0].contents[1].styleAttrs["http://www.w3.org/ns/ttml#styling fontSize"],
-                                {"unit": "em", "value": 0.105}
-                        );
+                assert.deepEqual(
+                    doc.body.contents[0].contents[1].styleAttrs["http://www.w3.org/ns/ttml#styling fontSize"],
+                    {"unit": "em", "value": 0.105}
+                );
 
-                        assert.deepEqual(
-                                doc.body.contents[0].contents[2].styleAttrs["http://www.w3.org/ns/ttml#styling fontSize"],
-                                {"unit": "px", "value": 10.5}
-                        );
+                assert.deepEqual(
+                    doc.body.contents[0].contents[2].styleAttrs["http://www.w3.org/ns/ttml#styling fontSize"],
+                    {"unit": "px", "value": 10.5}
+                );
 
-                        assert.deepEqual(
-                                doc.body.contents[0].contents[3].styleAttrs["http://www.w3.org/ns/ttml#styling fontSize"],
-                                {"unit": "c", "value": 0.105}
-                        );
+                assert.deepEqual(
+                    doc.body.contents[0].contents[3].styleAttrs["http://www.w3.org/ns/ttml#styling fontSize"],
+                    {"unit": "c", "value": 0.105}
+                );
 
-                    }
-            );
+            }
+        );
 
-        }
+    }
 );
 
+QUnit.test(
+    "Metadata Callbacks",
+    function (assert) {
+
+        var count = 0;
+        var cur_tag = 0;
+
+        mh = {
+            onOpenTag: function (ns, name, attrs) {
+
+                switch (count) {
+                    case 0:
+                        assert.equal(name, "title");
+                        assert.equal(ns, "http://www.w3.org/ns/ttml#metadata");
+                        cur_tag = 1;
+                        break;
+                    case 3:
+                        assert.equal(name, "conformsToStandard");
+                        assert.equal(ns, "urn:ebu:metadata");
+                        cur_tag = 2;
+                        break;
+                    case 4:
+                        assert.equal(name, "image");
+                        assert.equal(ns, "http://www.smpte-ra.org/schemas/2052-1/2010/smpte-tt");
+                        assert.equal(attrs[" imagetype"].value, "PNG");
+                        cur_tag = 3;
```

Comment:
> Why is there a whitespace in attrs[" image type"]?

`@imagetype` is not prefixed.

From https://www.w3.org/TR/xml-names/#defaulting : _the namespace name for an unprefixed attribute name always has no value_.

> Would also be nice to have the id of the image extracted in the unit test, since it is not obvious to me whether it will be available with namespace or not.

Done.

> Then finally have the image data extracted using and the first characters, or its length compared to the data would provide another pedagogical step on how to extract the actual image data.

Done.